nats.go

```diff
@@ -1083,14 +1086,11 @@ func (nc *Conn) processMsg(data []byte) {
 
 	sub.mu.Lock()
 
-	// This is a catch all for more than max messages delivered.
-	if sub.max > 0 && sub.msgs > sub.max {
-		sub.mu.Unlock()
-		nc.removeSub(sub)
-		nc.mu.Unlock()
-		return
-	}
-
```

Comment:
Deleted block here shows race condition. Why are we comparing sub.msgs here vs. sub.delivered? Why do this here, we DO NOT KNOW the message is going to be delivered. What if unsubscribe/etc... is called again, what if max is changed, etc...